custom_dc/README.md

```diff
@@ -0,0 +1,262 @@
+
+## Prerequisites
+
+- Install [docker](https://www.docker.com/products/docker-desktop/)
+- [Optional] Install [gcloud](https://cloud.google.com/sdk/docs/install-sdk)
+  command line tool.
+- [Optional] Have a Google Cloud Project that can be used to deploy the custom
+  instance.
+
+## API Key
+
+- A custom Data Commons needs to connect with main Data Commons. Get API key for
+  Data Commons by sending an email to `support@datacommons.org`.
+
+- [Optional] Provision a Google Maps API key from your GCP project. This is
+  optional and used for place search in visualization tools. Refer to [Maps API
+  Key](TODO) section for detailed instructions.
+
+## Local Development
+
+Custom Data Commons instance can be developed and tested locally by following
+the instructions below:
+
+### Build docker image locally
+
+```bash
+docker build --tag datacommons-website-compose:latest \
+-f build/web_compose/Dockerfile \
+-t website-compose .
+```
+
+### Run custom Data Commons locally with SQLITE database
+
+In the root of this repository, run:
+
+```bash
+docker run -it \
+--env-file $PWD/custom_dc/sqlite_env.list \
+-p 8080:8080 \
+-v $PWD/custom_dc/sample:/sqlite \
+-v $PWD/server/templates/custom_dc:/workspace/server/templates/custom_dc \
+datacommons-website-compose:latest
+```
+
+This brings up a local instance with sample data that are stored under
+`custom_dc/sample` folder.
+
+Now you can open `localhost:8080/tools/timeline` to browse these sample data.
+Also note the base Data Commons data are also avaiable in this instance.
+
+## Cloud Development and Deployment
+
+Custom Data Commons can be ran on the cloud as a production service. The SQLite
+approach can still be used. Note you need to copy the data folders into the
+cloud disk and mount it to the docker container then specify the enviornment
+variables based on the requirements of the cloud providers.
+
+We have also provided a specific deployment setup on Google Cloud Run that is
+based on Cloud SQL.
+
+### Setup Google Cloud SQL
+
+Create a Google Cloud SQL instance from the [Cloud
+Console](https://console.cloud.google.com/sql/instances). Set instance ID as
+`dc-graph`, choose the type as "MySQL" and create a database `datacommons`. You
+will need to set a user and password. Record the instance connection name in the
+form of "<project>:<region>:dc-graph"
+
+### Setup Google Cloud Storage
+
+Google Cloud Storage is used to hold the data CSV files. From the [Cloud
+Console](https://console.cloud.google.com/storage/browser), create a new bucket
+or pick an existing bucket and upload the data CSV files there. It's recommended
+to create intermediate folders for the files for easier management.
+
+### Upload Data Files
+
+Upload the data CSV files into GCS and record the folder path as
+`gs://<bucket-name>/.../`. You can start wit the sample data provided under
+`custom_dc/sample` and update to your own data later.
+
+### Run Custom Data Commons Locally with CloudSQL Database
```

Comment:
Is this just for testing?  Or is this a realistic production scenario?

Reply:
This is for testing against CloudSQL

Reply:
Then maybe we could make that clear with a "Testing: " prefix in title?